fili-sql/src/main/java/com/yahoo/bard/webservice/sql/presto/DruidQueryToPrestoConverter.java

```diff
@@ -0,0 +1,296 @@
+// Copyright 2019 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.sql.presto;
+
+import static com.yahoo.bard.webservice.druid.model.DefaultQueryType.GROUP_BY;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.druid.model.aggregation.Aggregation;
+import com.yahoo.bard.webservice.druid.model.orderby.LimitSpec;
+import com.yahoo.bard.webservice.druid.model.orderby.SortDirection;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.druid.model.query.DruidQuery;
+import com.yahoo.bard.webservice.druid.model.query.GroupByQuery;
+import com.yahoo.bard.webservice.sql.ApiToFieldMapper;
+import com.yahoo.bard.webservice.sql.DruidQueryToSqlConverter;
+import com.yahoo.bard.webservice.sql.aggregation.DruidSqlAggregationConverter;
+import com.yahoo.bard.webservice.sql.aggregation.SqlAggregation;
+import com.yahoo.bard.webservice.sql.evaluator.FilterEvaluator;
+import com.yahoo.bard.webservice.sql.evaluator.HavingEvaluator;
+import com.yahoo.bard.webservice.sql.helper.CalciteHelper;
+import com.yahoo.bard.webservice.sql.helper.SqlTimeConverter;
+import com.yahoo.bard.webservice.table.SqlPhysicalTable;
+
+import com.google.common.base.CaseFormat;
+import com.google.common.collect.ImmutableList;
+
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.sql.fun.SqlStdOperatorTable;
+import org.apache.calcite.tools.RelBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+/**
+ * Default implementation of converting a {@link DruidQuery} into a sql query.
+ */
+public class DruidQueryToPrestoConverter extends DruidQueryToSqlConverter {
```

Comment:
I think there's a lot of duplication here we can get rid of

Reply:
Discuss in person

Reply:
Fixed